### PR TITLE
Let student know if annotation is available 

### LINKED ISSUE #3025: Make sure student sees pdf annoations
1) If a pdf has annotations, make that clear to the student with a badge and a message "you have feedback from grader" or soemthing.

2) When indicating if a student has viewed the grade, let's add a check to see if they opened the pdf, if the pdf has annotations.

### PR DESCRIPTION
**Pull request by saxocellphone**
closes #3025.
Now the date of when student viewed annotation is stored in the database. Currently, it's displayed nowhere, so that's up for discussion. Also a note will appear in the TAResultbox if an annotation is available for the student.

### CODE REVIEW

File: migration/migrations/course/20181112133026_student_annotation.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+def up(config, conn, semester, course):
+    with conn.cursor() as cursor:
+        cursor.execute("ALTER TABLE gradeable_data ADD COLUMN gd_user_annotation_viewed_date timestamp(6) with time zone DEFAULT NULL;")
+
+
+def down(config, conn, semester, course):
+    with conn.cursor() as cursor:
+            cursor.execute("ALTER TABLE gradeable_data DROP COLUMN eg_regrade_request_date;")
```
Comment (KevinMackenzie) [inline migration/migrations/course/20181112133026_student_annotation.py]: You might want to check the column you're dropping : )

File: site/app/controllers/pdf/PDFController.php [hunk 2]
```diff
@@ -71,6 +71,9 @@ private function showStudentPDF(){
                 }
             }
         }
+        $ta_graded_gradeable = $graded_gradeable->getTaGradedGradeable();
```
Comment (KevinMackenzie) [inline site/app/controllers/pdf/PDFController.php]: The result of `getTaGradedGradeable` can be null, so you should check that.  Also noticed that there are a lot of unchecked results above in the function.  Don't need to change those for this PR, but something you might want to more gracefully handle. (i.e. `tryGetGradeable` will return `false` on failure)

**Comment by bmcutler** [comment]
@saxocellphone please resolve conflicts

**Comment by bmcutler** [comment]
@KevinMackenzie  please re-review

**Comment by bmcutler** [comment]
Thanks for the contributions but....
We are closing this since it's quite stale and needs alot of work.  It will be easier to start over.

**Pull request closed without merging**